src/com/google/enterprise/adaptor/DocumentContentTransform.java

```diff
@@ -0,0 +1,68 @@
+package com.google.enterprise.adaptor;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The document content transform can modify the content of a document.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class DocumentContentTransform extends OutputStream {
+
+  protected final Map<String, String> config;
+  protected final Metadata metadata;
+  protected final String contentType;
+  private OutputStream originalStream;
+
+  /**
+   * Constructs a document content transformer.
+   *
+   * @param config         the configuration for this instance
+   *                       (never null, but maybe empty)
+   * @param originalStream the original content stream
+   * @param contentType    the content type of the stream
+   *                       (could possibly be null or empty)
+   * @param metadata       the metadata already collected for this document
+   *                       (never null, but maybe empty)
+   */
+  public DocumentContentTransform(final Map<String, String> config,
+                                  final OutputStream originalStream, final String contentType,
+                                  final Metadata metadata) {
+    if (null == originalStream) {
+      throw new NullPointerException();
+    }
+    this.config = Collections.unmodifiableMap(new HashMap<String, String>(config));
+    this.originalStream = originalStream;
+    this.contentType = contentType;
+    this.metadata = metadata;
+  }
+
+  @Override
+  public void write(int b) throws IOException {
+    originalStream.write(b);
+  }
+
+  @Override
+  public void write(byte[] b) throws IOException {
+    originalStream.write(b);
+  }
+
+  @Override
+  public void write(byte[] b, int off, int len) throws IOException {
+    originalStream.write(b, off, len);
+  }
+
+  @Override
+  public void flush() throws IOException {
+    originalStream.flush();
+  }
+
+  @Override
+  public void close() throws IOException {
+    originalStream.close();
+  }
+}
```

Comment:
I've got a CL that will be changing DocumentHandler
<https://codereview.appspot.com/261770043/>.

I'll wait until we've submitted your content transform
CL before submitting the one I've got going.  That way
I'll get to resolve any emergent merge conflict.

Thank you,
PJ




- technology's compounding interest
-

On Fri, Aug 14, 2015 at 10:56 AM, PJ <pjo@google.com> wrote:

> Great.  Thank you.
>
> I've left four comments for you.
> 1 is important, about newInstance call needing to be in createPipeline
>
> 1 is thinking about the future to let transform change content-type
> variable
>
> 2 are requests for rewording comments.
>
> Thank you very much,
> PJ
>
>
>
>
> - technology's compounding interest
> -
>
> On Fri, Aug 14, 2015 at 12:54 AM, Dominik Weidenfeld <
> notifications@github.com> wrote:
>
>> In src/com/google/enterprise/adaptor/DocumentContentTransform.java
>> <https://github.com/googlegsa/library/pull/7#discussion_r37056075>:
>>
>> > +
>> > +  @Override
>> > +  public void write(byte[] b, int off, int len) throws IOException {
>> > +    originalStream.write(b, off, len);
>> > +  }
>> > +
>> > +  @Override
>> > +  public void flush() throws IOException {
>> > +    originalStream.flush();
>> > +  }
>> > +
>> > +  @Override
>> > +  public void close() throws IOException {
>> > +    originalStream.close();
>> > +  }
>> > +}
>>
>> done
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/googlegsa/library/pull/7/files#r37056075>.
>>
>
>